### PR TITLE
Add support for "COPY dist/ref tables FROM" progress report

### PR DESCRIPTION
**Pull request by onderkalaci**
Simply call Postgres' function to report the progress on
each row recieved.

We do not support this feature for append partitioned distributed
tables.

Note that we currently do not support "COPY dist/ref TO .." progress
report nicely. Citus has some specialized logic to support
"COPY dist/ref TO .." such that it either converts the underlying
command into "COPY (SELECT * FROM dist/ref ) ..." or sends COPY
command to shards directly. In the former case, "tuples_processed"
is only updated when the executor returns all the tuples, so the
progress is not accurate. In the latter case, Citus can actually
implement the progress report. But, for the sake of consistency,
we prefer to not implement at all.

### CODE REVIEW

File: src/backend/distributed/commands/multi_copy.c [hunk 1]
```diff
@@ -560,7 +561,11 @@ CopyToExistingShards(CopyStmt *copyStatement, QueryCompletionCompat *completionT
 
 		dest->receiveSlot(tupleTableSlot, dest);
 
-		processedRowCount += 1;
+		++processedRowCount;
+
+#if PG_VERSION_NUM >= PG_VERSION_14
+		pgstat_progress_update_param(PROGRESS_COPY_TUPLES_PROCESSED, processedRowCount);
```
Comment (onderkalaci) [inline src/backend/distributed/commands/multi_copy.c:567]: what about local copy?

Comment (onderkalaci) [reply src/backend/distributed/commands/multi_copy.c:567]: That is not a concern, we only need to consider COPY on the distributed tables.

File: src/backend/distributed/commands/multi_copy.c [hunk 2]
```diff
@@ -560,7 +561,11 @@ CopyToExistingShards(CopyStmt *copyStatement, QueryCompletionCompat *completionT
 
 		dest->receiveSlot(tupleTableSlot, dest);
 
-		processedRowCount += 1;
+		++processedRowCount;
+
+#if PG_VERSION_NUM >= PG_VERSION_14
```
Comment (SaitTalhaNisanci) [inline src/backend/distributed/commands/multi_copy.c:566]: What about `CopyToNewShards`? I thought that we needed this logic there as well but with an `append` table I am able to see the progress. Do you know why?

Comment (onderkalaci) [reply src/backend/distributed/commands/multi_copy.c:566]: With append, `BeginCopyFrom` sets the total_bytes. However, `tuples_processed` should be set by us. It is a single line change, so pushed that as well

Comment (onderkalaci) [reply src/backend/distributed/commands/multi_copy.c:566]: And, bytes_processed is done in a lower level APIs that we already call: `NextCopyFrom ->NextCopyFromRawFields ->CopyReadLine ->CopyReadLineText ->CopyLoadInputBuf ->CopyLoadRawBuf `

File: src/backend/distributed/commands/multi_copy.c (hunk 1 above)
Comment (onderkalaci) [reply src/backend/distributed/commands/multi_copy.c:567]: pg_stat_progress_copy doesn't work well for local execution as we re-start many COPY commands in the same backend where the state is cleared for each internal copy: https://github.com/postgres/postgres/blob/609b0652af00374b89411ea2613fd5bb92bca92c/src/backend/commands/copyfrom.c#L1206-L1217

**Comment by SaitTalhaNisanci** [comment]
`tuples_excluded` is not a concern for us I guess as we don't support WHERE clause. Should we create a small issue with COPY PROGRESS limitations so that when asked, we can show it the users?

**Comment by onderkalaci** [comment]
> tuples_excluded is not a concern for us I guess as we don't support WHERE clause. Should we create a small issue with COPY PROGRESS limitations so that when asked, we can show it the users?


I added [this](https://github.com/citusdata/citus/pull/5218/files#diff-270fb8edbd5d02a921c6b61d0f122e94fdc6b250d8617ada396427fe488b5170R2996-R3004) comment to the code, but feel free to open an issue (which we might already have?)

**Review by SaitTalhaNisanci — APPROVED** [review]
Minor: Can include the pg commit in the description or commit message

**Pull request merged**